codus/codelog
# Interactive change file

## Objective
In this PR, I I tried to solve #28, adding a functionality to the codelog, that permits the users to create the change files in an interactive way.

If the user add the `--interactive`(or just `-i`) option to the `new` command, the system will ask for the changes, accordingly to the responses from the user.

![interactive](https://user-images.githubusercontent.com/32220858/36952220-8666c5a8-1feb-11e8-8679-13a47631024c.gif)

The example above generated a file with the following content:

![image](https://user-images.githubusercontent.com/32220858/36952105-44b05710-1fea-11e8-92b2-df7801decd8e.png)

## Notes
* Currently, the command does not verify if the provided type is valid.
* The generated file is not exactly like the template. I just used the parser, provided by the `YAML` library.

## How to test
* Generate a change file, using the command
   ```
   codelog new -i
   ```
* Choose a type of change
* Add some changes
   * To increase the indent, add `:` to the end of the phrase
* Verify if the generated file is correct

## Checklist



[CONTRIBUTING]: https://github.com/codus/codelog/blob/master/CONTRIBUTING.md
